R/classes-plate_builder.R

```diff
@@ -203,18 +203,89 @@ convert_dilutions_to_numeric <- function(dilutions) {
 }
 
 #' @description
-#' Function translates sample names to sample types
-#' The function uses regular expressions to match the sample names
-#' to the sample types
+#' Function translates sample names to sample types based on the sample name
+#' from Luminex file and the sample name from the layout file, which may not
+#' be provided. The function uses regular expressions to match the sample names
+#' to the sample types.
 #'
-#' @param sample_names A vector of sample names
-#' @return A vector of sample types
+#' It parses the names as follows:
+#' If `sample_names` or `sample_names_from_layout` equals to `BLANK`, `BACKGROUND` or `B`,
+#' then SampleType equals to `BLANK`
+#' If `sample_names` or `sample_names_from_layout` equals to `STANDARD CURVE`,
+#' `SC`, `S` contains substring `1/\d+` and has prefix ` `, `S_`, `S `,
+#' `S` or `CP3`, then SampleType equals to `STANDARD CURVE`
+#' If `sample_names` or `sample_names_from_layout` equals to `NEGATIVE CONTROL`, `N`,
+#' or contains substring `NEG`, then SampleType equals to `NEGATIVE CONTROL`
+#' If `sample_names` or `sample_names_from_layout` starts with `P` followed by
+#' whitespace, `POS` followed by whitespace, `B770` or `10/190`
```

Comment:
I don't think that B770 or 10/190 is generic enough. I think that those names represent a set (or sample) used for the positive control. I think that the positive control are created in a way: "sample_name dilution" as in the example here: `B770 1/400`

Reply:
So just firstly parse the negative control an then use regex for positive control parsing